pyrealm/splash/splash.py

```diff
@@ -266,10 +283,10 @@ def estimate_daily_water_balance(
         # Check day_idx inputs to map either the single time index given in day_idx or
         # the whole dataset.
         if day_idx is None:
-            check_input_shapes(previous_wn, self.pn)
+            check_input_shapes(previous_wn, shape=self.shape)
             didx: int | slice = slice(self.shape[0])
```

Comment:
Same as above, I think - this would be much better as an explicit test and raise, rather my original subversion of the `check_input_shapes` process.